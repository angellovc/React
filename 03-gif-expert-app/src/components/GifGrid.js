import GifGridItem from './GifGridItem';
import useFetchGifs from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {
  const {data:images , loading} = useFetchGifs(category);
  return (
    <div className="card-grid">
    <h2 className="animate__animated animate__fadeIn">{category}</h2>
        {loading && <p>Loading...</p>}

        {(category !== '' && images[0] === 'Not Found') &&  <p>Not Found</p>}
        {images[0] === 'Not Found' && delete images[0]} 

        {
          images.map((image) => (
            <GifGridItem
              key={image.id}
              {...image}
          />))
        }
    </div>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid
