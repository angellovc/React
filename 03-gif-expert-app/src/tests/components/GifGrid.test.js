import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('<GifGrid /> Tests', () => {
    const category = 'Hello';

    test('Snapshot sould match', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Must display intems once images have been loaded', () => {
        useFetchGifs.mockReturnValue({
            data: [{
                id: '123',
                title: 'abc',
                url: 'http://localhost'
            }],
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(1);
    });

});