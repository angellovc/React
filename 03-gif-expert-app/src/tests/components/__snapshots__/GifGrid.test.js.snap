// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> Tests Snapshot sould match 1`] = `
<div
  className="card-grid"
>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Hello
  </h2>
  <p>
    Loading...
  </p>
</div>
`;
